### DISCUSSION TITLE
Allow for folders that do not affect generated route/url

### DISCUSSION THREAD

**Discussion by jsve** [withastro/roadmap#662] [Proposal]
# Summary

Similar to how exclusions work in `/pages` (https://docs.astro.build/en/core-concepts/routing/#excluding-pages) there could also be a way to have "silent" folders, to help organise pages. Such folders could for instance be prefixed with `#`.

# Background & Motivation

In a project where you have a large number of pages, for example a blog, it becomes difficult to keep them all in one folder. Introducing folders today also affects the routes to the pages. Having a way to organise the files / pages while still having "flat" urls would be helpful.

## Use case

**You want to separate the posts you are working on from the ones that are already published** 

With the introduction of "silent" folders you could keep the posts you are working on in `/blog/#drafts/` and all the other files flat in `/blog/`. You could also choose to organise the posts by year or year/month to keep the number of files per folder manageable (i.e. `/blog/#2023/#08/`).

For blogs that already choose to include the date in the path, eg. `/blog/2023/some-blog-post`, the issue is less relevant.

**You have multiple people publishing to the same blog**

With the introduction of "silent" folders you could have one folder per author. By utilising this you could also add limitations in who is allowed to commit changes to the respective folder (obviously out of scope for this RFC) through GitHub Actions or similar.

# Example
The bullets below illustrate the relationship between folders/files and generated routes after the introduction of "silent" folders:

 - file: `/blog/2023/some-blog-post` -> url: `/blog/2023/some-blog-post`
 - file: `/blog/#2023/some-blog-post` -> url: `/blog/some-blog-post`
 - file: `/blog/_2023/some-blog-post` -> url: `EXCLUDED` (https://docs.astro.build/en/core-concepts/routing/#excluding-pages)

# Goals

- Add a way to ignore folder beginning with the string `#` in the generated paths when creating routes (probably added somewhere around [here]( https://github.com/withastro/astro/blob/f224c539c880add0490b442b4429bdf73b58f6ff/packages/astro/src/core/util.ts#L165))

**Comment by florian-lefebvre** [comment]
Duplicate of https://github.com/withastro/roadmap/discussions/480
